tests/rustdoc/source-code-pages/keyword-macros.rs

```diff
@@ -0,0 +1,19 @@
+// This test ensures that keywords which can be followed by values (and therefore `!`)
+// are not considered as macros.
+// This is a regression test for <https://github.com/rust-lang/rust/issues/148617>.
+
+#![crate_name = "foo"]
+
+//@ has 'src/foo/keyword-macros.rs.html'
+
+fn a() {
```

Comment:
Maybe add at least one test case where the `!` is not separated by whitespace from the keyword?
e.g. `if! true{}`
To make sure it works, and doesn't regress in the future.

Also, do we have tests for `!`s following punctuation? e.g. something like `const ARR: [u8; 2] = [!0,! 0];`?

Reply:
The same issue cannot happen with something other than idents but added it just in case.